test/unit/error.spec.ts

```diff
@@ -292,3 +285,88 @@ test("return from nested finally", () => {
     `;
     expect(util.transpileAndExecute(code)).toBe("finally AB");
 });
+
+test("throw and catch custom error object", () => {
+    util.testFunction`
+        try {
+            throw { x: "Hello error object!" };
+        } catch (error) {
+            return error.x;
+        }
+    `.expectToMatchJsResult();
+});
+
+test.each(["Error", "RangeError", "ReferenceError", "SyntaxError", "TypeError", "URIError"])(
+    "throw builtin Errors as classes",
+    errorType => {
+        util.testFunction`
+            try {
+                throw new ${errorType}("message")
+            } catch (error) {
+                if (error instanceof Error) {
+                    return \`\${error}\`;
```

Comment:
It would be cleaner to just do `error.toString()` which is also transformed to `tostring(error)`